modules/spellfiles.nix

```diff
@@ -0,0 +1,175 @@
+{
+  lib,
+  config,
+  pkgs,
+  ...
+}:
+let
+  inherit (lib) types;
+
+  spellfilesHashes = import ../generated/spellfiles.nix;
+
+  availableSpellfiles = lib.attrNames spellfilesHashes;
+
+  # en.utf-8.spl -> en
+  extractLanguage = filename: builtins.head (builtins.split "\\." filename);
+
+  availableLanguages = lib.pipe availableSpellfiles [
+    (map extractLanguage)
+    lib.unique
+  ];
+
+  cfg = config.spellfiles;
+in
+{
+  options.spellfiles = {
+    enable = lib.mkOption {
+      type = types.bool;
+      default = false;
+      example = true;
+      description = ''
+        Whether to automatically fetch and install vim spellfiles (`.spl` and `.sug`).
+
+        If enabled, Nixvim will automatically populate `spell/en.utf-8.spl`-type files by fetching
+        them from [the official vim FTP server](https://ftp.nluug.nl/pub/vim/runtime/spell/).
+
+        This module allows choosing which spellfiles to install.
+      '';
+    };
+
+    languages = lib.mkOption {
+      type = with types; either (enum [ "auto" ]) (listOf (enum availableLanguages));
+      default = "auto";
+      example = [
+        "en"
+        "fr"
+      ];
+      description = ''
+        By default, spellfiles from the languages listed in `opts.spelllang` will be fetched (or
+        `"en"` if `opts.spelllang` is not defined).
+
+        Set to `[ ]` to disable this behavior.
+      '';
+    };
+
+    includeSuggestions = lib.mkOption {
+      type = types.bool;
+      default = false;
+      example = true;
+      description = ''
+        Whether to also fetch suggestion files (`.sug`) when available.
+
+        Warning, using suggestion files can lead to higher memory usage.
+
+        See https://neovim.io/doc/user/spell.html#spell-NOSUGFILE.
+      '';
+    };
+
+    spellfiles = lib.mkOption {
+      type = with types; listOf (enum availableSpellfiles);
+      default = [ ];
+      description = ''
+        Explicitly specify spellfiles to install.
+
+        Possible values correspond to the spellfiles available at https://ftp.nluug.nl/pub/vim/runtime/spell/.
+      '';
+      example = [
+        "en.utf-8.spl"
+        "en.utf-8.sug"
+        "fr.utf-8.spl"
+      ];
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    extraFiles =
+      let
+        /*
+          Infers the list of language ids from `opts.spelllang`:
+
+          Turns: "en_us,fr,de"
+          into: [ "en" "fr" "de" ]
+          # https://neovim.io/doc/user/spell.html#spell-load
+        */
+        languagesFromSpelllang = lib.pipe (config.spelllang or "en") [
+
+          # "en_us,fr,de" -> [ "en_us" "fr" "de" ]
+          (lib.splitString ",")
+
+          # Remove region from language id ("en_us" -> "en")
+          (map (lang: lib.head (builtins.match "^([^_]+)" lang)))
```

Comment:
This is doing the same thing as L15, but with a slightly different approach.

Could we decide on a preferred approach and introduce a common function?

Reply:
Ah my mistake, I think I must've skimmed over the comment and thought it was doing the same thing.